src/inmanta/config.py

```diff
@@ -301,9 +301,8 @@ def get(self) -> T:
             has_deprecated_option = cfg.has_option(self.predecessor_option.section, self.predecessor_option.name)
             has_new_option = cfg.has_option(self.section, self.name)
             if has_deprecated_option and not has_new_option:
-                warnings.warn(
-                    "Config option %s is deprecated. Use %s instead." % (self.predecessor_option.name, self.name),
-                    category=DeprecationWarning,
```

Comment:
There are quite some deprecation warnings. I wonder if it might be worth creating a `class InmantaDeprecationWarning(InmantaWarning, DeprecationWarning)`? If so we might want to have `CompilerDeprecationWarning` inherit from it. The inheritance graph would become a diamond but I think that should be fine.

Reply:
Yes. I though about that. I didn't do it because I was afraid that this might derail into a full mirror on the Python warnings hierarchy. But I can still add it.

Reply:
It sounds like a valid concern. Exception hierarchy is something I generally feel very uncomfortable with, I find it very difficult to say which is best. The main motivation for making it a differentiable exception class would be if we ever want to allow more customization. Perhaps we should leave it as it is now until such a use case arises.